src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -655,6 +684,15 @@ def engine_forward(
             callback = context.get("callback")
             stop = context.get("stop")
 
+            max_new_tokens = generation_config.max_new_tokens
+            if max_new_tokens:
+                max_tokens = max_new_tokens + len(generated_tokens)
+            else:
+                max_tokens = generation_config.max_length
+                max_tokens = (
+                    max_tokens if max_tokens > 0 else (100 * self.sequence_length)
+                )
```

Comment:
> `(100 * self.sequence_length)`

I see this was copied from before but I think just `self.sequence_length` (or 2x) should be plenty? Not sure what `max_tokens` means at this layer so a comment would help

Reply:
will touchbase with Damian offline about this and open up a new PR